test-program to check -race is supported makes OBS builds fail

The tiny program test introduced in 7188b1c3e22dc4aec016635add6b1a4a6e62ae46 to test if -race is supported did not fix the issue on Ubuntu and continues to make the Ubuntu OBS build fail. 

[   87s] # command-line-arguments
[   87s] runtime.raceinit: __tsan_init: not defined
[   87s] runtime.raceinit: __tsan_map_shadow: not defined
[   87s] runtime.racefini: __tsan_fini: not defined
[   87s] runtime.racemapshadow: __tsan_map_shadow: not defined
[   87s] runtime.racemalloc: __tsan_malloc: not defined
[   87s] runtime.racegostart: __tsan_go_start: not defined
[   87s] runtime.racegoend: __tsan_go_end: not defined
[   87s] runtime.raceacquireg: __tsan_acquire: not defined
[   87s] runtime.racereleaseg: __tsan_release: not defined
[   87s] runtime.racereleasemergeg: __tsan_release_merge: not defined
[   87s] runtime.racefingo: __tsan_finalizer_goroutine: not defined
[   87s] runtime.RaceDisable: __tsan_go_ignore_sync_begin: not defined
[   87s] runtime.RaceEnable: __tsan_go_ignore_sync_end: not defined
[   87s] runtime.raceread: __tsan_read: not defined
[   87s] runtime.racereadpc: __tsan_read_pc: not defined
[   87s] runtime.racewrite: __tsan_write: not defined
[   87s] runtime.racewritepc: __tsan_write_pc: not defined
[   87s] runtime.racereadrange: __tsan_read_range: not defined
[   87s] runtime.racereadrangepc1: __tsan_read_range: not defined
[   87s] runtime.racewriterange: __tsan_write_range: not defined
[   87s] runtime.racewriterangepc1: __tsan_write_range: not defined
[   87s] /usr/lib/go-1.6/pkg/tool/linux_amd64/link: too many errors
[   87s] can't load package: package .: no buildable Go source files in /usr/src/packages/BUILD
[   87s] Makefile:5250: recipe for target 'check-proxy' failed
[   87s] make[1]: *** [check-proxy] Error 1
[   87s] make[1]: Leaving directory '/usr/src/packages/BUILD'
[   87s] dh_auto_test: make -j1 check returned exit code 2
[   87s] debian/rules:5: recipe for target 'build' failed
[   87s] make: *** [build] Error 2
[   87s] dpkg-buildpackage: error: debian/rules build gave error exit status 2
[   87s] 
[   87s] build80 failed "build cc-oci-runtime.dsc" at Thu Feb 16 04:49:38 UTC 2017.
[   87s] 
[   87s] ### VM INTERACTION START ###
[   90s] ### VM INTERACTION END ###
[   90s] 
[   90s] build80 failed "build _service:set_version:cc-oci-runtime.dsc" at Thu Feb 16 04:49:41 UTC 2017.
[   90s]

--- a/Makefile.am
+++ b/Makefile.am
@@ -276,23 +276,9 @@ cc_proxy_extra_dist =			\
 
 CHECK_DEPS += check-proxy
 
-define CONTENT
-package main
-import "fmt"
-func main() {
-	fmt.Printf("Executing test program")
-}
-endef
-export CONTENT
-
 check-proxy:
-	$(AM_V_GEN)go_test_extra_args="" ; \
-	echo "$$CONTENT">./racetest.go ; \
-	if go build -race ./racetest.go; then \
-		go_test_extra_args+="-race" ; \
-	fi ; \
-	rm ./racetest* ; \
-	go test -v "$$go_test_extra_args" -timeout 2s $(srcdir)/proxy
+	$(AM_V_GEN)proxy_test_common_args="-v -timeout 2s $(srcdir)/proxy" ; \
+	go test -race $$proxy_test_common_args || go test $$proxy_test_common_args
 
 check-go:
 	@$(top_srcdir)/.ci/ci-go-static-checks.sh

